backend/compact-connect/tests/app/test_transaction_monitoring.py

```diff
@@ -0,0 +1,222 @@
+         'Resource': 'arn:${Token[AWS.Partition.8]}:states:::sqs:sendMessage.waitForTaskToken',
+         'TimeoutSeconds': 1123200,
+         'Type': 'Task'}
+
+         Notice sometimes the token is embedded in a string, and sometimes it is a standalone value.
+         We need to account for both cases here.
+
+        the dynamic token numbers are removed from the string so the tests are stable.
+        """
+        # this matches the token pattern so we can replace the numbers
+        token_pattern = re.compile(r'\$\{Token\[([^\]]+)\]\}')
+
+        def replace_tokens(value):
+            if isinstance(value, dict):
+                return {k: replace_tokens(v) for k, v in value.items()}
+            if isinstance(value, list):
+                return [replace_tokens(v) for v in value]
+            if isinstance(value, str):
+
+                def strip_numbers(match):
+                    token_content = match.group(1)
+                    parts = token_content.split('.')
+                    stripped_parts = [part for part in parts if not part.isdigit()]
+                    return '${Token[' + '.'.join(stripped_parts) + ']}'
+
+                return token_pattern.sub(strip_numbers, value)
+            return value
+
+        return replace_tokens(definition)
+
+    def test_workflow_generate_process_transaction_history_lambda_with_permissions(self):
+        transaction_monitoring_stack = self.app.pipeline_stack.prod_stage.transaction_monitoring_stack
+        transaction_monitoring_stack_template = Template.from_stack(transaction_monitoring_stack)
+
+        lambda_properties = self.get_resource_properties_by_logical_id(
+            transaction_monitoring_stack.get_logical_id(
+                transaction_monitoring_stack.compact_state_machines[
+                    'aslp'
+                ].transaction_processor_handler.node.default_child
+            ),
+            transaction_monitoring_stack_template.find_resources(CfnFunction.CFN_RESOURCE_TYPE_NAME),
+        )
+
+        self.assertEqual('handlers.transaction_history.process_settled_transactions', lambda_properties['Handler'])
+
+        handler_role_logical_id = transaction_monitoring_stack.get_logical_id(
+            transaction_monitoring_stack.compact_state_machines[
+                'aslp'
+            ].transaction_processor_handler.role.node.default_child
+        )
+        # get the policy attached to the role using this match
+        # "Roles": [
+        #     {
+        #         "Ref": "<role logical id>"
+        #     }
+        # ]
+        policy = next(
+            policy
+            for policy_logical_id, policy in transaction_monitoring_stack_template.find_resources(
+                'AWS::IAM::Policy'
+            ).items()
+            if handler_role_logical_id in policy['Properties']['Roles'][0]['Ref']
+        )
+
+        # We need to ensure the lambda can only read the secret for its compact
+        self.assertIn(
```

Comment:
Great choice to test for this statement in particular. Since you're doing that, you might as well test each configured compact, rather than just `aslp`. You have the actual `Stack` so you can get each configured compact with `transaction_monitoring_stack.node.get_context('compacts')`, then iterate over each, rather than picking one.